test/e2e/plain_provisioner_test.go

```diff
@@ -1029,6 +1030,158 @@ var _ = Describe("plain provisioner bundleinstance", func() {
 	})
 })
 
+var _ = Describe("plain provisioner garbage collection", func() {
+	When("a Bundle has been deleted", func() {
+		var (
+			ctx context.Context
+			b   *rukpakv1alpha1.Bundle
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the testing Bundle resource")
+			b = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "e2e-ownerref-bundle-valid",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Source: rukpakv1alpha1.BundleSource{
+						Type: rukpakv1alpha1.SourceTypeImage,
+						Image: &rukpakv1alpha1.ImageSource{
+							Ref: "testdata/bundles/plain-v0:valid",
+						},
+					},
+				},
+			}
+			Expect(c.Create(ctx, b)).To(BeNil())
+
+			By("eventually reporting an Unpacked phase")
+			Eventually(func() (string, error) {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(b), b); err != nil {
+					return "", err
+				}
+				return b.Status.Phase, nil
+			}).Should(Equal(rukpakv1alpha1.PhaseUnpacked))
+		})
+		AfterEach(func() {
+			By("deleting the testing Bundle resource")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, b))).To(BeNil())
```

Comment:
Note: client.IgnoreNotFound as we're deleting the Bundle within the individual It spec(s).